docs/dev/tutorials/custom-paymaster-tutorial.md

```diff
@@ -1,72 +1,101 @@
-- Read the [gas estimation for transaction](../developer-guides/transactions/fee-model.md#gas-estimation-during-a-transaction-for-paymaster-and-custom-accounts) guide.
-- If you haven't, please refer to the first section of the [quickstart tutorial](../building-on-zksync/hello-world.md).
+The tutorial code is available [here](https://github.com/matter-labs/custom-paymaster-tutorial).
 
-## Installing dependencies
+## Set up the project
 
-We will use the zkSync hardhat plugin for developing this contract. Firstly, we should install all the dependencies for it:
-
-::: code-tabs
-
-@tab:active folder
+1. Create a project folder and `cd` into it:
 
 ```sh
 mkdir custom-paymaster-tutorial
 cd custom-paymaster-tutorial
 ```
 
-@tab:active yarn
+2. Initialize the project:
 
-```bash
+```sh
 yarn init -y
-yarn add -D typescript ts-node ethers@^5.7.2 zksync-web3 hardhat @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy
 ```
-:::
 
-::: tip
+3. Add the project dependencies, including Hardhat and all zkSync packages:
 
-The current version of `zksync-web3` uses `ethers v5.7.x` as a peer dependency. An update compatible with `ethers v6.x.x` will be released soon.
+```sh
+yarn add -D typescript ts-node ethers@^5.7.2 zksync-web3 hardhat @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy @matterlabs/zksync-contracts @openzeppelin/contracts @openzeppelin/contracts-upgradeable
+```
 
+::: tip
+- The current version of `zksync-web3` uses `ethers v5.7.x` as a peer dependency. 
+- An update compatible with `ethers v6.x.x` will be released soon.
 :::
 
-Since we are working with zkSync contracts, we also need to install the package with the contracts and its peer dependencies:
+4. Create the required folders.
 
 ```sh
-yarn add @matterlabs/zksync-contracts @openzeppelin/contracts @openzeppelin/contracts-upgradeable
+mkdir contracts deploy
 ```
 
-Then create the `hardhat.config.ts` config file, `contracts` and `deploy` folders, like in the [quickstart tutorial](../building-on-zksync/hello-world.md).
-
-::: tip
+5. Create the file `hardhat.config.ts` and add the following, replacing the `ethNetwork` with an RPC URL:
 
-You can use the zkSync CLI to scaffold a project automatically. Find [more info about the zkSync CLI here](../../api/tools/zksync-cli/)
+```ts
+import "@matterlabs/hardhat-zksync-deploy";
+import "@matterlabs/hardhat-zksync-solc";
+
+module.exports = {
+  zksolc: {
+    version: "1.3.8",
+    compilerSource: "binary",
+    settings: {},
+  },
+  defaultNetwork: "zkSyncTestnet",
+
+  networks: {
+    zkSyncTestnet: {
+      url: "https://testnet.era.zksync.dev",
+      ethNetwork: "goerli", // the RPC URL of the network (e.g. `https://goerli.infura.io/v3/<API_KEY>`)
+      zksync: true,
+    },
+  },
+  solidity: {
+    version: "0.8.8",
+  },
+};
+```
 
+::: tip
+- Use [npx hardhat](https://hardhat.org/hardhat-runner/docs/getting-started#quick-start) to scaffold a default Hardhat project.
```

Comment:
I'd not advise people to use `npx hardhat`. At least with zkSync CLI you get all our dependencies but `npx hardhat` give you nothing that works with zkSync out of the box

Reply:
OK, can you make the suggestion and I'll accept it?

Reply:
DONE